src/consumerauthorization/5_0_0/sd/authorization-token/authorization-token_sd.tex

```diff
@@ -0,0 +1,364 @@
+
+\textbf{Simple token}s are not holding any kind of information. These can only be verified by consulting the system implementing the authorizationToken service. See the \textit{verify} operation.
+
+\textbf{Self-contained token}s are holding all the necessary information for the provider system to verify it independently. If an encryption key associated with the targeted service provider system is stored, the token will be provided encrypted with that key. See the \textit{register-encryption-key} operation.
+
+\fsubsection{verify}{AuthorizationTokenVerifyRequest}{AuthorizationTokenVerifyResponse}{}{ErrorResponse}
+
+Operation \textit{verify} checks whether a given simple type token is valid or not, is associated with the requester (provider) system or not and it returns the belonged token details. If the token is verified, then based on the returned token details, the provider system can ensure that the token is valid to the targeted service-operation or not.
+
+Verifying self-contained type of tokens with this operation is not possible.
+
+\fsubsection{get-public-key}{Identity}{PublicKeyResponse}{}{ErrorResponse}
+
+Operation \textit{get-public-key} returns the \pref{PublicKey} of the implementing system, if available. Having a \pref{PublicKey} is only possible if the system supports an asymmetric encryption solution. Some self-contained token types also support asymmetric encryption, allowing the requester (provider) system to verify whether the token was created by the same system that the \pref{PublicKey} belongs to.
+
+\fsubsection{register-encryption-key}{AuthorizationEncryptionKeyRegistrationRequest}{AuthorizationEncryptionKeyRegistrationResponse}{}{ErrorResponse}
+
+Operation \textit{register-encryption-key} saves and stores a \pref{String} key and an encryption algorithm identifier, that is provided by and belongs to the requester (provider) system. Any time when a self-contained token is generated that is associated with the same provider, the token will be provided to the consumer system encrypted, using the specified key and encryption algorithm. If the given algorithm is using any addition (besides the encryption key) for the encryption process, such as salt or initialization vector for example, then this addition is returned to the requester (provider) system. Once a consumer system sends a service request with an encrypted self-signed token, the provider system can decrypt it with the encryption key defined by itslef and with the addition received as the response of this operation if any.
+
+\fsubsection{unregister-encryption-key}{Identity}{OperationStatus}{}{ErrorResponse}
+
+Operation \textit{unregister-encryption-key} removes the stored encryption key and algorithm identifier associated with the requester (provider) system. The result of this operation is that further tokens generated to any of the provider's service, won't be encrypted. 
+
+\clearpage
+
+\section{Information Model}
```

Comment:
Ordering should be:

1. AuthorizationTokenGenerationRequest
2. Identity
3. AccessTokenScope
4. AuthorizationTokenGenerationResponse
5. ErrorResponse
6. AuthorizationTokenVerifyRequest
7. AuthorizationTokenVerifyResponse
8. PublicKeyResponse
9. AuthorizationEncryptionKeyRegistrationRequest
10. AuthorizationEncryptionKeyRegistrationResponse